_analyzers/stemming.md

```diff
@@ -0,0 +1,163 @@
+
+[`kstem`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/kstem/): A lightweight stemmer designed for English that combines algorithmic stemming with a built-in dictionary. It reduces plurals to singulars, converts verb tenses to their base forms, and removes common derivational endings. 
+
+
+[`stemmer`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/stemmer/): Provides algorithmic stemming for various languages, including English, with options for different stemming algorithms like `light_english`, `minimal_english`, and `porter2`. 
+
+
+[`snowball`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/snowball/): Utilizes the Snowball algorithm to provide efficient and accurate stemming for multiple languages, including English, French, German, and others. 
+
+### Dictionary stemmers
+
+Dictionary stemmers rely on extensive dictionaries to map words to their root forms, effectively handling irregular words. They look up each word in a precompiled list to find its corresponding stem. This operation is more resource intensive but often yields better results for irregular words and words which might appear to have a similar stem but are very different in their meaning.
+
+The most prominent example of a dictionary stemmer is [`hunspell`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/hunspell/) token filter, which leverages Hunspell, a spell checker engine used in many open-source applications.
+
+### Consideration
+
+- Algorithmic Stemmers are suitable when processing speed and memory efficiency are priorities, and the language has relatively regular morphological patterns.
+- Dictionary Stemmers are ideal when accuracy in handling irregular word forms is crucial, and resources are available to support the increased memory usage and processing time.
+
+Choose algorithmic stemmers for speed and efficiency with regular languages. Opt for dictionary stemmers for accuracy with complex languages, acknowledging the increased resource requirements.
+{: .note}
+
+### Additional stemming configuration
+
+Although "organize" and "organic" share a common linguistic root, leading a stemmer to produce "organ" for both, their conceptual differences are significant. In practical search scenarios, this shared root can lead to irrelevant matches.
+
+Further stemming configuration can address these challenges, using the following methods:
+
+- **Explicit Stemming Overrides**: Rather than relying solely on algorithmic stemming, you can define specific rules. Using [`stemmer_override`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/stemmer-override/) allows you to make sure that "organize" remains unchanged, while "organic" is reduced to "organ." This grants granular control over the final form of terms.
```

Comment:
```suggestion
- **Explicit stemming overrides**: Rather than relying solely on algorithmic stemming, you can define specific stemming rules. Using [`stemmer_override`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/stemmer-override/) allows you to ensure that "organize" remains unchanged, while "organic" is reduced to "organ." This grants granular control over the final form of terms.
```